lib/hls/hls_parser.js

```diff
@@ -2414,11 +2414,10 @@ shaka.hls.HlsParser = class {
     const contentMimeType = response.headers['content-type'];
 
     if (!contentMimeType) {
-      throw new shaka.util.Error(
-          shaka.util.Error.Severity.CRITICAL,
-          shaka.util.Error.Category.MANIFEST,
-          shaka.util.Error.Code.HLS_COULD_NOT_GUESS_MIME_TYPE,
```

Comment:
You should also remove this error code.  See `lib/util/error.js`, look for "RETIRED", and copy the format of those comments.  You will also need to run the tests and update any failing tests.

Reply:
@joeyparrish Done, tests in progress. I just had a general question, you already knew how to fix this issue, the changes were pretty small so why didn't you finish solving it? I was just curious like this is your project and you are striving to make it perfect and this issue was left unsolved for 2 months. was this intentional for beginners to solve or you were caught up with other works that you missed.

Reply:
There's a lot to unpack in this comment.  Let me come back to answering this later today when I have more time to compose a more detailed reply.

Reply:
ok, Thank you :-)